### DISCUSSION TITLE
Structure Plan for Website Rework

### DISCUSSION THREAD

**Discussion by Flyingmana** [OpenMage/magento-lts#3771] [General]
over the last few years there were repeated requests to rework the website, or event to remove it full.

While adding/removing single pages is simple, its probably also time to approach this with a bit more professionalism by first defining the Use-cases and Audiences we can think of.

In the second step(later) we can then evaluate how these can impact our project Goals, and how we can best convert our visitors into a win for the Project.

I maintain the Plan at https://github.com/OpenMage/organizational/blob/main/website_plan.md , but for simplicity I will copy it here for the start.

> ## Target Usecases / Target Audiences
> 
> Several usecases we found so far. 
> Might we need "Landing Pages" for each of them? So we can have the main page focused for marketing the Project, and direct most people easily to dedicated Pages where they find easier what they are looking for,
> 
> ### existing users looking for support and developers
> 
> relevant Pages:
> 
> * list of freelancers
> * list of agencies
> * list of hosters?
> 
> relevant links (probably need a dedicated page to put into context):
> 
> * links to stackexchange, github discussions, discord 
> * links to websites for small jobs? (I dont know how they are called)
> 
> relevant search Terms?
> 
> * 
> * 
> 
> ### learning about future plans, focus, roadmap
> 
> relevant Pages
> 
> * Focus Page with established Project Goals ( https://github.com/OpenMage/organizational/blob/main/Goals.md )
> * Roadmap listing already decided new Changes for upcoming Versions (probably only containing the ones we have at least 50% confidence of finishing in time)
> * blog with semi regular posts about sub projects we are kicking off (for example removing an outdated js library, creating a new Theme, PHP Version updates)
> 
> ### staying updated about new changes and achivements in the project
> 
> summary of the most relevant changelog entries, could probably get covered by the Blog
> 
> While Devs can check directly the Changelog, for more business focused people and even more sales people a simplified summary is a benefit. Could then also get a bit decorated with marketing speach.
> 
> ### undecided people finding out what we can offer for new Setups and Shops
> 
> Thats a possibly problematic point, as we lack in features and integrations in many areas.
> Highlighting our strengths would probably cover only the low cost in hosting and maintenance?
> 
> ### new people/devs wanting to try out the project
> 
> * link to demo page
> * short/simple tutorial to setup the project on local
> * extended tutorial for a best practice setup
> * tutorial for non-devs?
> 
> ### people wanting to contribute to the project
> 
> We do have a link to the Github Page, and also a Contributor Guideline. But besides the main repository there is so much more we do need support with. It might be good to have a dedicated landing page giving an overview.
> 
> 
> 
> ## other usecases / Flows
> 
> ### A place to highlight supporting companies?
> 
> ### showcase some of the unique Features we have to shine?

**Comment by fballiano** [comment]
In the past I've suggested to remove the website altogether and redirect to the github repo, but I now think that https://github.com/OpenMage/magento-lts/pull/3659 could be a good replacement for the website. many many projects use that layout so people are used to it, there's no "design" so nothing new to learn in order to add new pages, which is the most important thing to keep it updated.

**Reply by Flyingmana** [reply]
Moving all into a single repository is not incompatible with making a Plan of what kind of pages we actually need and for which audiences they are intended.
What is your Opinion regarding the Plan for the Pages we need, is there anything you think we dont need, or which we miss?

**Comment by sreichel** [comment]
For the demo site ...

- add a cron job that 
  -  installs the latest version
  - installs sample data
  - restrict default admin account permissions from harmful actions
 
 (maybe use n98-magerun for adding demo-admin account and other sttings ...)

**Comment by DegrizNet** [comment]
I had some time today so I made a mockup how I see page for OpenMage :)

![openmage](https://github.com/OpenMage/magento-lts/assets/5030172/584ac0a1-3c11-4e35-ae68-389697fc7f1f)

**Reply by Flyingmana** [reply]
It is looking good, but you seem to be already 2 steps ahead of me.

What is your opinion about the Content and audience Plan?
In my Plan I mentioned the part about finding "Freelancers", how would you describe the "customer" journey from this Page for someone looking for Freelancers?
And do you think we actually need such a Page, or should we omit it because its not a relevant Audience we should serve?

**Reply by DegrizNet** [reply]
My opinion is that the current content is only for developers and is missing a large part of the audience. Also, from a marketing perspective, there are some things that I truly don't like. That is the reason for doing a mockup, as it was to show how I think it would be better.

There are a couple of different audiences, and I think the site would need pages to take care of each of these audiences. For example:

- New developers: They need a list of modules and resources on how to not only install OpenMage but also set up tax for their country, etc. If they have a simple way to learn everything, they will promote OpenMage and probably purchase modules or services for specific tasks.
- Payment/shipping providers, etc.: They need reasons why OpenMage is interesting for them and their clients, so they need a page that gives them all info and success examples. Currently, as I hear, they think Magento 1.9 is dead, and they are not upgrading modules but removing working ones and support pages. If they see there is a community and if they are going to develop modules for OpenMage, they would get new clients from OpenMage. I think it would be much easier for them to invest in it. But for OpenMage to have the upper hand, it needs customer database (emailing, social media, site visits, etc.).
- Module providers: They also need insurance and promotions to keep developing modules, so a directory is needed. But it is important to have a filter on this listing so only verified modules are there.
- New shop owners: Reasons why to choose, also listing of freelancers and modules, etc.
And then we have pages for existing shop owners with info on how to migrate/upgrade and so on. There should be multiple pages. I am a little limited with time, so I will stop here for now. :)
About freelancers, first, I would start only with a list of OpenMage contributors. So if you contribute to OpenMage, you are on the list. Later on, the best way would be a system like they have here https://www.codeable.io/.

Without some system/listing like this, new shop owners will probably get some "developer" without real knowledge and with a product that will be below their expectations.

**Reply by DegrizNet** [reply]
Ok I just want to add if there is interes I can help work on website rework as contribution to OpenMage project.. it would be good if we would talk more in depth about this and define everything but then I can also code design into theme on OpenMage with all modules necessery.

**Comment by addison74** [comment]
When I see how slow the other solutions are, including Magento 2, I'm glad I'm working with a fossil. Around the beginning of 2010, Magento was considered extremely slow and today with the new hardware resources it runs extremely fast. In Windows WSL + DDEV + Docker amazes those who used WAMP and XAMPP in the past. I haven't used Debian + Virtualmin for testing, I only do it before moving everything to production.

OpenMage really worth a shot folks!

**Reply by DegrizNet** [reply]
Just the day before, I conducted a mockup test to assess the speed of a template I'm planning to use. I managed to achieve a score of 100 on both mobile and desktop platforms (view the report (https://pagespeed.web.dev/analysis/https-demo-degriz-net-si-addict-lip-glow-oil/5uurzf4bb3?form_factor=mobile). However, I've noticed a significant challenge – many individuals are opting for alternative platforms such as Shopify and WooCommerce due to their perceived cost-effectiveness and reliability. Even if this perception isn't entirely accurate, it's the prevailing sentiment. Given that my livelihood relies on ecommerce projects, I'm currently contemplating a switch from Openmage/Magento 2 to something else if things don't improve. :(

**Reply by addison74** [reply]
Magento 2 has 1700 Issues reported and 900 opened PR. These figures increased dramatically in the last years. Issues, expensive extensions and custom code, decreased the interest for this platform. This is my personal opinion.

**Reply by DegrizNet** [reply]
I know, but Magento 2 has shown significant improvement with version 2.4.6 compared to a few versions ago. However, the challenge lies in the fact that most purchased modules encountered issues during the upgrade process. Almost all modules faced functionality problems due to changes related to PHP8 and the removal of Zend, among other modifications. This resulted in a substantial amount of time being spent on issue resolution, consequently making updates more expensive.

Opting for OpenMage poses a similar challenge, as it can also be costly. 3rd party modules such as those for payment and delivery providers contribute to the overall expense. For my country's delivery services, they mentioned that developing an extension for OpenMage is not considered worthwhile, given the limited interest from stores. The majority have shifted to platforms like WooCommerce and Shopify. They even indicated that developing a Magento 2 module wasn't deemed worthwhile.

**Reply by DegrizNet** [reply]
I think OpenMage site should have strong customer database to influence this providers to develop modules for plaftorm. Also there shuld be listing/marketplace for all available options on site, developers listing with certification / credibility rating, huge knowledge database,.. the list is long. :)

**Reply by Flyingmana** [reply]
@DegrizNet 
> However, I've noticed a significant challenge – many individuals are opting for alternative platforms such as Shopify and WooCommerce due to their perceived cost-effectiveness and reliability. 

what section should we have on our website, to target this topic? 
In the Plan it is probably then targeted at the audience "undecided people finding out what we can offer for new Setups and Shops", right?

> I think OpenMage site should have strong customer database to influence this providers to develop modules for plaftorm

As I understand this, you recommend us to add "module providers" to the list of targeted audiences in the Plan?

**Reply by DegrizNet** [reply]
Hmm, for sections, I think there should be some brainstorming on what to add to avoid having too much. :) I would create multiple pages for different audiences. It's better to have more than less.

Also, I think it would be great to do everything on OpenMage. I built my site, and it is really easy to manage. For example:

Categories are landing pages.
Products are module listings.
Attribute pages for agencies, developers, etc. (open source by Fishpig).
And a blog conenctedwith related products/ categories (open source by Fishpig).
And probably some custom things.

**Reply by empiricompany** [reply]
>  I managed to achieve a score of 100 on both mobile and desktop platforms (view the report

awesome, let's do it! We immediately need a modern and high-performance frontend and an official website, I agree with you on the target audience

**Comment by kiatng** [comment]
I like @DegrizNet mock up very much. It's comprehensive. It has pretty much everything and covers different audiences. One section or link (perhaps in Footer:Case Studies or a separate link) to add in the mock up is for those looking to migrate from Magento 1 and 2, and other shopping carts. We should have an honest opinions if OpenMage is a right fit for these shop owners.

**Reply by DegrizNet** [reply]
Yes, drop down is missing but I have planned it... I went over couple of major eccomerce platform pages and they all have case studys and so on.
